integrationTests/vm/arwen/testdata/c-api-tests/ecBenchmark/ecBenchmark.c

```diff
@@ -0,0 +1,461 @@
+typedef byte P521PRIVKEY[66];
+
+typedef byte P224MRESULT[57];
+typedef byte P256MRESULT[65];
+typedef byte P384MRESULT[97];
+typedef byte P521MRESULT[133];
+
+typedef byte P224MCRESULT[29];
+typedef byte P256MCRESULT[33];
+typedef byte P384MCRESULT[49];
+typedef byte P521MCRESULT[67];
+
+const int LIMIT = 1000;
+
+void init() 
+{
+}
+
+void initialVariablesAndCallsTest()
+{
+    int fieldOrderHandle = bigIntNew(0);
+    int basePointOrderHandle = bigIntNew(0);
+    int eqConstantHandle = bigIntNew(0);
+    int p224Handle = p224Ec();
+    int p224XBasePointHandle = bigIntNew(0);
+    int p224YBasePointHandle = bigIntNew(0);
+    ellipticCurveGetValues(p224Handle,fieldOrderHandle,basePointOrderHandle,eqConstantHandle,p224XBasePointHandle,p224YBasePointHandle);
+    int p256Handle = p256Ec();
+    int p256XBasePointHandle = bigIntNew(0);
+    int p256YBasePointHandle = bigIntNew(0);
+    ellipticCurveGetValues(p256Handle,fieldOrderHandle,basePointOrderHandle,eqConstantHandle,p256XBasePointHandle,p256YBasePointHandle);
+    int p384Handle = p384Ec();
+    int p384XBasePointHandle = bigIntNew(0);
+    int p384YBasePointHandle = bigIntNew(0);
+    ellipticCurveGetValues(p384Handle,fieldOrderHandle,basePointOrderHandle,eqConstantHandle,p384XBasePointHandle,p384YBasePointHandle);
+    int p521Handle = p521Ec();
+    int p521XBasePointHandle = bigIntNew(0);
+    int p521YBasePointHandle = bigIntNew(0);
+    ellipticCurveGetValues(p521Handle,fieldOrderHandle,basePointOrderHandle,eqConstantHandle,p521XBasePointHandle,p521YBasePointHandle);
+}
+
+void addEcTest()
+{
+    int fieldOrderHandle = bigIntNew(0);
+    int basePointOrderHandle = bigIntNew(0);
+    int eqConstantHandle = bigIntNew(0);
+    int p224Handle = p224Ec();
+    int p224XBasePointHandle = bigIntNew(0);
+    int p224YBasePointHandle = bigIntNew(0);
+    ellipticCurveGetValues(p224Handle,fieldOrderHandle,basePointOrderHandle,eqConstantHandle,p224XBasePointHandle,p224YBasePointHandle);
+    int p256Handle = p256Ec();
+    int p256XBasePointHandle = bigIntNew(0);
+    int p256YBasePointHandle = bigIntNew(0);
+    ellipticCurveGetValues(p256Handle,fieldOrderHandle,basePointOrderHandle,eqConstantHandle,p256XBasePointHandle,p256YBasePointHandle);
+    int p384Handle = p384Ec();
+    int p384XBasePointHandle = bigIntNew(0);
+    int p384YBasePointHandle = bigIntNew(0);
+    ellipticCurveGetValues(p384Handle,fieldOrderHandle,basePointOrderHandle,eqConstantHandle,p384XBasePointHandle,p384YBasePointHandle);
+    int p521Handle = p521Ec();
+    int p521XBasePointHandle = bigIntNew(0);
+    int p521YBasePointHandle = bigIntNew(0);
+    ellipticCurveGetValues(p521Handle,fieldOrderHandle,basePointOrderHandle,eqConstantHandle,p521XBasePointHandle,p521YBasePointHandle);
+
+    int i;
+    for (i = 0; i < LIMIT/4; i++)
+    {
+        addEC(p224XBasePointHandle,p224YBasePointHandle,p224Handle,p224XBasePointHandle,p224YBasePointHandle,p224XBasePointHandle,p224YBasePointHandle);
+    }
```

Comment:
you can eventually extract this for in a function. not the most pleasant thing seeing it repeated 4 times

Reply:
you can try something like this with a lambda: https://stackoverflow.com/questions/61329542/how-to-pass-a-lambda-function-as-a-parameter